diffrax/_term.py

```diff
@@ -776,3 +785,136 @@ def _to_vjp(_y, _diff_args, _diff_term):
+
+    $d \mathbf{v}_t = - \gamma \mathbf{v}_t dt - u
+    \nabla f( \mathbf{x}_t ) dt + \sqrt{2 \gamma u} d W_t.$
+    """
+
+    gamma: _LangevinX
+    u: _LangevinX
+    control: AbstractBrownianPath
+
+    def vf(self, t: RealScalarLike, y: _LangevinTuple, args: Args) -> _LangevinX:
+        x, v = y
+
+        def _fun(_gamma, _u, _v):
+            return jnp.sqrt(2 * _gamma * _u) * jnp.ones(_v.shape, _v.dtype)
+
+        d_v = jtu.tree_map(_fun, self.gamma, self.u, v)
+        return d_v
+
+    def contr(self, t0: RealScalarLike, t1: RealScalarLike, **kwargs) -> _LangevinBM:
+        return self.control.evaluate(t0, t1, **kwargs)
+
+    def prod(self, vf: _LangevinX, control: _LangevinX) -> _LangevinTuple:
+        if isinstance(control, AbstractBrownianIncrement):
+            control = control.W
+
+        dv = vf
+        dw = control
+        v_out = jtu.tree_map(operator.mul, dv, dw)
+        x_out = jtu.tree_map(jnp.zeros_like, v_out)
+        return x_out, v_out
+
+
+def _broadcast_pytree(source, target_tree_shape):
+    # Requires that source is a prefix tree of target_tree_shape
+    def _inner_broadcast(_src_arr, _inner_tree_shape):
+        _arr = jnp.asarray(_src_arr)
+        return jtu.tree_map(
+            lambda _leaf: jnp.broadcast_to(_arr, _leaf.shape), _inner_tree_shape
+        )
+
+    return jtu.tree_map(_inner_broadcast, source, target_tree_shape)
+
+
+class _LangevinDriftTerm(AbstractTerm):
+    gamma: _LangevinX
+    u: _LangevinX
+    grad_f: Callable[[_LangevinX], _LangevinX]
+
+    def vf(self, t: RealScalarLike, y: _LangevinTuple, args: Args) -> _LangevinTuple:
+        x, v = y
+        f_x = self.grad_f(x)
+        d_x = v
+        d_v = jtu.tree_map(
+            lambda _gamma, _u, _v, _f_x: -_gamma * _v - _u * _f_x,
+            self.gamma,
+            self.u,
+            v,
+            f_x,
+        )
+        d_y = (d_x, d_v)
+        return d_y
+
+    def contr(self, t0: RealScalarLike, t1: RealScalarLike, **kwargs) -> RealScalarLike:
+        return t1 - t0
+
+    def prod(self, vf: _LangevinTuple, control: RealScalarLike) -> _LangevinTuple:
+        return jtu.tree_map(lambda _vf: control * _vf, vf)
+
+
+def make_langevin_term(
+    gamma: PyTree[ArrayLike],
+    u: PyTree[ArrayLike],
+    grad_f: Callable,
+    bm: AbstractBrownianPath,
+    x0: _LangevinX,
+) -> MultiTerm[tuple[_LangevinDriftTerm, _LangevinDiffusionTerm]]:
+    r"""Creates a term that represents the Underdamped Langevin Diffusion, given by:
+    $d \mathbf{x}_t = \mathbf{v}_t dt$
+    $d \mathbf{v}_t = - \gamma \mathbf{v}_t dt - u
+    \nabla f( \mathbf{x}_t ) dt + \sqrt{2 \gamma u} d W_t.$
+    where $\mathbf{x}_t, \mathbf{v}_t \in \mathbb{R}^d$ represent the position
+    and velocity, $W$ is a Brownian motion in $\mathbb{R}^d$,
+    $f: \mathbb{R}^d \rightarrow \mathbb{R}$ is a potential function, and
+    $ \gamma,u\in\mathbb{R}^{d\times d}$ are diagonal matrices governing
+    the friction and the dampening of the system.
```

Comment:
Can you check whether these render correctly in the docs? (I suspect you'll need to a couple of newlines and things.)

Reply:
You're right, thanks!

Also I have an issue with my docs. For some reason the class headers are not appearing. So where you have `::: diffrax.ODETerm` the docs don't display "ODETerm" but just its docstring. Do you know how to fix this? I googled around but couldn't find anything.